File: C#/StringCalculator/StringCalculator/Calculator.cs
namespace StringCalculator
{
	public class Calculator
	{
		// uncomment to start!

		//private readonly IOutputWriter _outputWriter;

		//public Calculator(IOutputWriter outputWriter)
		//{
		//	_outputWriter = outputWriter;
		//}

		public int Add(string numbers)
		{
			var sum = 0;

			if (numbers == string.Empty)
			{
				//_outputWriter.Write(sum);
				return sum;
			}

			var splitNumbers = numbers.Split(",");

			foreach (var number in splitNumbers)
			{
				if (int.TryParse(number, out var integerNumber))
				{
					sum += integerNumber;
				}
			}

			//_outputWriter.Write(sum);
			return sum;
		}
	}
}

File: Java/src/main/java/com/codurance/stringcalculator/Calculator.java
package com.codurance.stringcalculator;

public class Calculator {
//    private OutputWriterInterface outputWriter;
//
//    public Calculator(OutputWriterInterface outputWriter)
//    {
//    	outputWriter = outputWriter;
//    }

    public int add(String numbers) {
        int sum = 0;

        if (numbers.equals("")) {
            //outputWriter.Write(sum);
            return sum;
        }

        String[] splitNumbers = numbers.split(",");

        for (String number : splitNumbers) {
            int integerNumber = Integer.parseInt(number);
                sum += integerNumber;
        }

        //outputWriter.Write(sum);
        return sum;
    }
}
